internal/cluster/cluster.go

```diff
@@ -59,15 +73,20 @@ func NewCluster() (Cluster, error) {
 	return cluster, nil
 }
 
-func (c *Cluster) Name() string {
-	// Handle not named clusters
-	for _, prop := range c.Cib.Configuration.CrmConfig.ClusterProperties {
-		if prop.Id == clusterNameProperty {
-			return prop.Value
-		}
+func getCorosyncAuthkey(corosyncKeyPath string) (string, error) {
+	kp, err := internal.Md5sum(corosyncKeyPath)
+	return kp, err
+}
+
+func getName() string {
+	keyStr, err := getCorosyncAuthkey(corosyncKeyPath)
+	if err != nil {
+		return ""
```

Comment:
I don't know if we should return an empty string here. The unique reason to have an error here is that the corosync key file doesn't exist, so maybe we should handle in a better way.
But maybe there is a reason to have it this way

PD: In any case, as this is executed after executing `getCorosyncAuthkey` already in the `NewCluster` it shouldn't never fail I guess

Reply:
noted!